modules/core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/BulkServiceImpl.java

```diff
@@ -328,4 +332,36 @@ public boolean await(Duration duration) throws InterruptedException {
                  .collect(Collectors.toList());
     }
 
+    @Override
+    public void appendExternalBucket(BulkBucket bucket) {
+        String commandId = bucket.getCommandId();
+
+        BulkCommand command = getCommand(commandId);
+        String stream = Framework.getService(BulkAdminService.class).getInputStream(command.getAction());
+
+        String key = commandId; // TODO do we keep a bucket number in the K/V to build this key?
```

Comment:
here we want to load balance processing so better commandId:counter counter can be a global atomic long